pages/overview/what-is-jsonschema.md

```diff
@@ -6,36 +6,285 @@ title: What is JSON Schema?
+    print("Invalid JSON object.")
+
+```
+Now we are dealing with a complex JSON structure that represents an order. The basic validation logic checks whether the required fields exist in the JSON object. However, as the structure becomes more complex, the validation code becomes more complicated and prone to errors. Moreover, this approach lacks support for checking data types, handling nested structures, and enforcing specific constraints.
+
+
+### JSON Schema to the rescue
+
+JSON Schema provides a solution to this problem. It is a specification language for JSON that allows you to describe the structure, content, and semantics of a JSON instance. With JSON Schema, you can define metadata about an object's properties, specify whether fields are optional or required, and define expected data formats.
+
+By using JSON Schema, people can better understand the structure and constraints of the JSON data they receive. It enables applications to validate data, ensuring it meets the defined criteria. With JSON Schema, you can make your JSON more readable, enforce data validation, and improve interoperability across different programming languages.
+
+Using the same example:
+```
+from jsonschema import validate
+
+# Using JSON Schema
+data = {
+    "product": {
+        "name": "Widget",
+        "price": 10.99,
+        "quantity": 5
+    }
+}
+
+schema = {
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "title": "Product",
+    "type": "object",
+    "properties": {
+        "product": {
+            "type": "object",
+            "properties": {
+                "name": {
+                    "type": "string"
+                    },
+                "price": {
+                    "type": "number", 
+                    "minimum": 0
+                    },
+                "quantity": {
+                    "type": "integer", 
+                    "minimum": 1
+                    }
+            },
+            "required": ["name", "price", "quantity"]
+        }
+    },
+    "required": ["product"]
+}
+
+try:
+    validate(data, schema)
+    print("Valid JSON object.")
+except Exception as e:
+    print("Invalid JSON object:", e)
+
+```
+
+In the above code snippet, we defined a JSON Schema that describes the expected structure and constraints of the JSON data. We use the `jsonschema` library to validate the `data` against the `schema`. If the data doesn't conform to the schema, an exception is raised, providing detailed information about the failure.
+
+By using JSON Schema, we can easily define and enforce constraints, making the validation process more robust and manageable. It improves the readability of the code and reduces the chances of data-related issues.
+
+
+### Steps to Validate JSON Data Using JSON Schema
+
+1. **Create the schema**
+
+    You begin by defining the rules in a JSON Schema document using [keywords](https://json-schema.org/draft/2020-12/json-schema-validation.html#name-validation-keywords-for-any).
```

Comment:
I wonder if it would be better to link to the docs here?
Being linked to a specification document for a new user can be daunting.

Reply:
That might be better for beginners. But it may also require users to do extra navigation to find the specific information regarding `keywords`. The direct link might be more efficient in guiding users to the specific content.

Or maybe if it provides both in a way like this: 
"You begin by defining the rules in a JSON Schema document using [keywords](), which are part of the broader set of specifications provided in the [JSON Schema documentation]()"
Linking both the docs and the specific section regarding keywords.

Please let me know what you think about this.

Reply:
The specification, getting started, and glossary pages are also being linked in the starting in the *Overview* section, which could help new users before going straight to the steps. I feel both options are good in their own way. 😅 

Please let me know your thoughts regarding this.

Reply:
Hi @Relequestual, could you please confirm about this and the below changes? I will proceed with updating the PR accordingly. 
Thankyou.

Reply:
This is an introduction for those unfamiliar with JSON Schema. For beginners. I don't think linking to the specification documents is appropriate so early.

If you feel we should additionally link to the keywords in the specifications, we must link to all of them (core and validation), at which point you should also explain there are multiple documents, and potentially even explain vocabularies... and I think it's too early for that.

Reply:
Of course, that makes more sense. I misunderstood before. 😅
Will make the changes accordingly.